src/models/stackedAreaChart.js

```diff
@@ -379,10 +389,25 @@ nv.models.stackedAreaChart = function() {
 
                 var xValue = xAxis.tickFormat()(chart.x()(singlePoint,pointIndex));
 
+                // Keeps track of the tooltip valueFormatter if the chart changes to expanded view
+                if (stacked.style() === 'expand' || stacked.style() === 'stack_percent') {
+                    if ( !oldValueFormatter ) {
+                        oldValueFormatter = interactiveLayer.tooltip.valueFormatter();
+                    }
+                    //Forces the tooltip to use percentage in 'expand' mode.
+                    interactiveLayer.tooltip.valueFormatter(yAxis.tickFormat(d3.format('%')));
+                }
+                else {
+                    if (oldValueFormatter) {
+                        interactiveLayer.tooltip.valueFormatter(oldValueFormatter);
+                        oldValueFormatter = null;
+                    }
+                }
+
                 //If we are in 'expand' mode, force the format to be a percentage.
                 var valueFormatter = (stacked.style() == 'expand') ?
                     function(d,i) {return d == null ? "N/A" : d3.format(".1%")(d);} :
-                    function(d,i) {return d == null ? "N/A" : yAxis.tickFormat()(d); };
+                    interactiveLayer.tooltip.valueFormatter();
```

Comment:
you mean combining it with this part?

             if (showYAxis) {
                 var ticks;
                 if (stacked.offset() === 'wiggle') {
                     ticks = 0;
                 }
                 else {
                     ticks = nv.utils.calcTicksY(availableHeight/36, data);
                 }
                 yAxis.scale(y)
                     ._ticks(ticks)
                     .tickSize(-availableWidth, 0);
 
                     if (stacked.style() === 'expand' || stacked.style() === 'stack_percent') {
                         if ( !oldYTickFormat )
                             oldYTickFormat = yAxis.tickFormat();
                         //Forces the yAxis to use percentage in 'expand' mode.
                         yAxis.tickFormat(d3.format('%'));
                     }
                     else {
                         if (oldYTickFormat) {
                             yAxis.tickFormat(oldYTickFormat);
                             oldYTickFormat = null;
                         }
                     }
 
                 g.select('.nv-y.nv-axis')
                     .transition().duration(0)
                     .call(yAxis);
             }